.github/workflows/buildAndTest.yml

```diff
@@ -266,21 +266,19 @@ jobs:
       # Run clang-tidy against only the changes. The 'clang-tidy-diff' script
       # does this if supplied with the diff.
       - name: clang-tidy
-        if: ${{ always() }}
```

Comment:
@teqdruid do you know why this was here? IIUC It means that the job would run even if cancelled (or if an earlier job failed). Is it just that we want to run the clang-tidy checks even if other checks fail? Should I add it back in?

https://docs.github.com/en/actions/learn-github-actions/expressions#always

Reply:
It was desirable to run the clang-tidy checks even when the check-circt test fails. Yeah, add it back.

Reply:
What about if I just moved the clang-tidy step to before the main build. I think I can do this if I run cmake before clang-tidy. Then, that clang-tidy won't run if an earlier step, such as checking out the source code, fails. As well, if the lint fails, we won't bother running the build. It might be leaner on resources.  I can just leave this as-is, too... I'm just looking around for anything we might want to change, while I'm here